numba/np/random/distributions.py

```diff
@@ -574,3 +574,164 @@ def random_logseries(bitgen, p):
             return 1
         else:
             return 2
+
+
+@register_jitable
+def random_binomial_btpe(bitgen, n, p):
+    r = min(p, 1.0 - p)
+    q = 1.0 - r
+    fm = n * r + r
+    m = int(fm)
+    p1 = int(2.195 * np.sqrt(n * r * q) - 4.6 * q) + 0.5
+    xm = m + 0.5
+    xl = xm - p1
+    xr = xm + p1
+    c = 0.134 + 20.5 / (15.3 + m)
+    a = (fm - xl) / (fm - xl * r)
+    laml = a * (1.0 + a / 2.0)
+    a = (xr - fm) / (xr * q)
+    lamr = a * (1.0 + a / 2.0)
+    p2 = p1 * (1.0 + 2.0 * c)
+    p3 = p2 + c / laml
+    p4 = p3 + c / lamr
+
+    case = 0
+    y = k = 0
+    while 1:
+        if case == 0:
+            nrq = n * r * q
+            u = next_double(bitgen) * p4
+            v = next_double(bitgen)
+            if (u > p1):
+                case = 1
+                continue
+            y = int(xm - p1 * v + u)
+            case = 6
+            continue
+        elif case == 1:
+            if (u > p2):
+                case = 2
+                continue
+            x = xl + (u - p1) / c
+            v = v * c + 1.0 - abs(m - x + 0.5) / p1
+            if (v > 1.0):
```

Comment:
So far I've been able to figure out the failure as follows:

The logic is not going through this branch, i.e. The logic jumps from case 0 to case 1 and is supposed to go through here but doesn't and ends up returning incorrect values in the next iteration. 

Probably we're overlooking some sort of precision issue over here.

Reply:
Is the issue in the expression `p1 = int(2.195 * np.sqrt(n * r * q) - 4.6 * q) + 0.5`, seems like in the NumPy source it's `p1 = floor(<same expr>)`, which has a different behaviour in the negative case.

Reply:
I corrected this, but doesn't seem like the issue at hand is being fixed.